FWCore/Framework/test/stubs/TestStreamAnalyzers.cc

```diff
@@ -368,92 +335,77 @@ namespace edmtest {
       void analyze(edm::Event const&, edm::EventSetup const&) override {
         ++m_count;
         ++(luminosityBlockCache()->value);
+        ++valueAccumulatedForStream_;
       }
 
       void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override {
-        bl = true;
-        el = false;
+        valueAccumulatedForStream_ = 0;
+        endLumiWasCalled_ = false;
       }
 
       static std::shared_ptr<Cache> globalBeginLuminosityBlock(edm::LuminosityBlock const& iLB,
                                                                edm::EventSetup const&,
                                                                RunContext const*) {
         ++m_count;
-        gbl = true;
-        gel = false;
+        globalBeginLumiCalled_ = true;
         auto pCache = std::make_shared<Cache>();
         ++(pCache->lumi);
         return pCache;
       }
 
-      static std::shared_ptr<Cache> globalBeginLuminosityBlockSummary(edm::LuminosityBlock const&,
-                                                                      edm::EventSetup const&,
-                                                                      LuminosityBlockContext const*) {
+      static std::shared_ptr<SummaryCache> globalBeginLuminosityBlockSummary(edm::LuminosityBlock const&,
+                                                                             edm::EventSetup const&,
+                                                                             LuminosityBlockContext const*) {
         ++m_count;
-        gbls = true;
-        gels = false;
-        bls = true;
-        els = false;
-        if (!gbl) {
+        if (!globalBeginLumiCalled_) {
           throw cms::Exception("begin out of sequence")
               << "globalBeginLuminosityBlockSummary seen before globalBeginLuminosityBlock";
         }
-        return std::make_shared<Cache>();
+        globalBeginLumiCalled_ = false;
```

Comment:
Before this PR, there were many more static variables in these stream test modules. I removed most of them in this PR. The static variables did not cause a problem before because the test modules were only used in one test configuration that set the number of concurrent lumis to 1 (and concurrent runs also to 1). 

I've added a new test that is actually running multiple lumis and multiple runs concurrently. I had to remove almost all the statics.

This particular static is OK because it is only used in globalBeginLuminosityBlock and globalBeginLuminosityBlockSummary. Those two transitions are still serial, basically occurring just after the input source lumi transition.

It will still fail if there is more than one instance of that module class type in the same test configuration. Both the old test and the new test in this PR do not do this. I suppose some future test configuration could and we could put the variable in a GlobalCache instead. I didn't because it's more work and complexity and not really necessary unless we plan to use this test module in new ways after this PR... If you request this, I could add this as part of this PR or as a separate PR later...

Reply:
This is the new test FWCore/Framework/test/testRunLumiCaches_cfg.py

Reply:
Thanks for the explanation, let's leave changes not really needed for this PR for later.